### PR TITLE
fix: [P4PU-469] Fix assistance form

### PR DESCRIPTION
**Pull request by vito80ba**
This PR refact the way to manage input types to avoid sync request to the service

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/components/AssistanceForm/AssistanceForm.tsx [hunk 1]
```diff
@@ -31,18 +29,26 @@ export const AssistanceForm = () => {
...
-      const { data: zendeskAssistance } = await utils.apiClient.token.getZendeskAssistanceToken({
-        userEmail: email
-      });
+      await utils.apiClient.token
+        .getZendeskAssistanceToken({
+          userEmail: email
+        })
+        .then((data) => {
+          const { data: zendeskAssistance } = data;
+          console.log(zendeskAssistance);
```
Comment (AnisLucidi01) [inline src/components/AssistanceForm/AssistanceForm.tsx:38]: can you remove this console log?

Comment (vito80ba) [reply src/components/AssistanceForm/AssistanceForm.tsx]: Right, thanks

**Review by AnisLucidi01 — APPROVED** [review]

File: src/components/AssistanceForm/AssistanceForm.tsx [hunk 2]
```diff
@@ -31,18 +29,25 @@ export const AssistanceForm = () => {
...
-      });
+      await utils.apiClient.token
+        .getZendeskAssistanceToken({
+          userEmail: email
+        })
+        .then((data) => {
+          const { data: zendeskAssistance } = data;
+          const resultData = zendeskAssistanceTokenResponseSchema.safeParse(zendeskAssistance);
+          if (!resultData.success) throw resultData.error;
+          if (zendeskAssistance.assistanceToken == '' || zendeskAssistance.returnTo == '')
```
Comment (mmoio) [inline src/components/AssistanceForm/AssistanceForm.tsx:40]: Should we check for whitespaces?
```javascript
if (!zendeskAssistance.assistanceToken.trim() || !zendeskAssistance.returnTo.trim())
```

File: src/components/AssistanceForm/AssistanceForm.tsx [hunk 3]
```diff
@@ -31,18 +29,25 @@ export const AssistanceForm = () => {
...
     e.preventDefault();
     try {
-      const { data: zendeskAssistance } = await utils.apiClient.token.getZendeskAssistanceToken({
-        userEmail: email
-      });
+      await utils.apiClient.token
+        .getZendeskAssistanceToken({
+          userEmail: email
+        })
+        .then((data) => {
```
Comment (mmoio) [inline src/components/AssistanceForm/AssistanceForm.tsx:36]: Can we remove the await and move the catch as `then(data => {}).catch(e => {})` or do the following instead?
```javascript
const data = await utils.apiClient.token.getZendeskAssistanceToken()
```

File: src/components/AssistanceForm/AssistanceForm.tsx (hunk 2 above)
Comment (vito80ba) [reply src/components/AssistanceForm/AssistanceForm.tsx:40]: No more. Now the BE has fix a old problem with white spaces :-)

**Review by mmoio — CHANGES_REQUESTED** [review]

**Pull request merged**